### ISSUE TITLE
Unable to deploy Flow app to IBM WebSphere Application Server 18.0 Liberty

### ISSUE THREAD

**Issue by Peppe** [vaadin/flow#3960]
Deploying a Vaadin app to Tomcat 8.0.x, 8.5.x and 9.0.x  being built with production mode has some visual elements broken.

## Steps to reproduce
1) Download Beverage buddy from https://vaadin.com/start/v10-simple-ui
1) Import it to Eclipse
1) Download WebSphere from https://developer.ibm.com/wasdev/downloads/#asset/runtimes-wlp-webProfile7 and extract it
1) Start it up with wls/bin/server run
1) Drop the war into wls/usr/servers/defaultServer/dropins
1) Access the app at http://localhost:9080/<name of your war>

## Broken parts
App deploys fine but when you try to access it with browser, you'll get a big error message. Same error in browser so you can't access the app.
```
peppe }> ~/Documents/Dev/Tools/wlp$ bin/server run
objc[23176]: Class JavaLaunchHelper is implemented in both /Library/Java/JavaVirtualMachines/jdk1.8.0_121.jdk/Contents/Home/jre/bin/java (0x102d284c0) and /Library/Java/JavaVirtualMachines/jdk1.8.0_121.jdk/Contents/Home/jre/lib/libinstrument.dylib (0x102df24e0). One of the two will be used. Which one is undefined.
Launching defaultServer (WebSphere Application Server 18.0.0.1/wlp-1.0.20.cl180120180309-2209) on Java HotSpot(TM) 64-Bit Server VM, version 1.8.0_121-b13 (en_US)
[AUDIT   ] CWWKE0001I: The server defaultServer has been launched.
[AUDIT   ] CWWKE0100I: This product is licensed for development, and limited production use. The full license terms can be viewed here: https://public.dhe.ibm.com/ibmdl/export/pub/software/websphere/wasdev/license/base_ilan/ilan/18.0.0.1/lafiles/en.html
[ERROR   ] CWWKF0042E: A feature definition cannot  be found for the  admincenter-1.0 feature.  Try running the command, bin/installUtility install admincenter-1.0,  to install the feature. Alternatively, you can run the command, bin/installUtility install defaultServer,  to install all features that are referenced by this configuration.
[AUDIT   ] CWWKZ0058I: Monitoring dropins for applications.
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://localhost:9080/testing/
[AUDIT   ] CWWKZ0001I: Application testing started in 5.145 seconds.
[AUDIT   ] CWWKF0012I: The server installed the following features: [jsp-2.3, ejbLite-3.2, managedBeans-1.0, servlet-3.1, jsf-2.2, beanValidation-1.1, ssl-1.0, jndi-1.0, appSecurity-2.0, jsonp-1.0, jdbc-4.1, jaxrs-2.0, el-3.0, jaxrsClient-2.0, json-1.0, jpaContainer-2.1, cdi-1.2, distributedMap-1.0, webProfile-7.0, websocket-1.1, jpa-2.1].
[AUDIT   ] CWWKF0011I: The server defaultServer is ready to run a smarter planet.
[err] [Default Executor-thread-10] INFO com.vaadin.flow.server.startup.ServletDeployer - Automatically deploying Vaadin servlet to /*
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[err] [Default Executor-thread-38] WARN com.vaadin.flow.server.DefaultDeploymentConfiguration - 
===========================================================
Vaadin is running in DEBUG MODE.
Add productionMode=true to web.xml to disable debug features.
To show debug window, add ?debug to your application URL.
===========================================================
[err] [Default Executor-thread-38] INFO com.vaadin.flow.server.DefaultDeploymentConfiguration - Will use webcomponents-loader.js polyfill discovered in webjars/webcomponentsjs
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[err] [Default Executor-thread-42] ERROR com.vaadin.flow.router.InternalServerError - There was an exception while trying to navigate to ''
[err] java.lang.IllegalArgumentException: Unable to create an instance of 'com.vaadin.starter.beveragebuddy.ui.views.reviewslist.ReviewsList'. The constructor threw an exception.
[err] 	at com.vaadin.flow.internal.ReflectTools.createProxyInstance(ReflectTools.java:507)
[err] 	at com.vaadin.flow.internal.ReflectTools.createInstance(ReflectTools.java:444)
[err] 	at com.vaadin.flow.di.DefaultInstantiator.getOrCreate(DefaultInstantiator.java:64)
[err] 	at com.vaadin.flow.di.Instantiator.createRouteTarget(Instantiator.java:157)
[err] 	at com.vaadin.flow.router.internal.AbstractNavigationStateRenderer.lambda$getRouteTarget$1(AbstractNavigationStateRenderer.java:114)
[err] 	at java.util.Optional.orElseGet(Optional.java:267)
[err] 	at com.vaadin.flow.router.internal.AbstractNavigationStateRenderer.getRouteTarget(AbstractNavigationStateRenderer.java:113)
[err] 	at com.vaadin.flow.router.internal.AbstractNavigationStateRenderer.handle(AbstractNavigationStateRenderer.java:165)
[err] 	at com.vaadin.flow.router.Router.handleNavigation(Router.java:198)
[err] 	at com.vaadin.flow.router.Router.navigate(Router.java:170)
[err] 	at com.vaadin.flow.router.Router.initializeUI(Router.java:92)
[err] 	at com.vaadin.flow.component.UI.doInit(UI.java:202)
[err] 	at com.vaadin.flow.server.BootstrapHandler.createAndInitUI(BootstrapHandler.java:988)
[err] 	at com.vaadin.flow.server.BootstrapHandler.synchronizedHandleRequest(BootstrapHandler.java:388)
[err] 	at com.vaadin.flow.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:40)
[err] 	at com.vaadin.flow.server.VaadinService.handleRequest(VaadinService.java:1496)
[err] 	at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:335)
[err] 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
[err] 	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1255)
[err] 	at [internal classes]
[err] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[err] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[err] 	at java.lang.Thread.run(Thread.java:745)
[err] Caused by: java.lang.reflect.InvocationTargetException
[err] 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[err] 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[err] 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[err] 	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[err] 	at com.vaadin.flow.internal.ReflectTools.createProxyInstance(ReflectTools.java:476)
[err] 	... 39 more
[err] Caused by: java.lang.IllegalStateException: Can't find resource 'frontend://src/views/reviewslist/reviews-list.html' via the servlet context
[err] 	at com.vaadin.flow.component.polymertemplate.DefaultTemplateParser.getTemplateContent(DefaultTemplateParser.java:111)
[err] 	at com.vaadin.flow.component.polymertemplate.TemplateDataAnalyzer.parseTemplate(TemplateDataAnalyzer.java:180)
[err] 	at com.vaadin.flow.component.polymertemplate.TemplateInitializer.<init>(TemplateInitializer.java:91)
[err] 	at com.vaadin.flow.component.polymertemplate.PolymerTemplate.<init>(PolymerTemplate.java:78)
[err] 	at com.vaadin.flow.component.polymertemplate.PolymerTemplate.<init>(PolymerTemplate.java:93)
[err] 	at com.vaadin.starter.beveragebuddy.ui.views.reviewslist.ReviewsList.<init>(ReviewsList.java:78)
[err] 	... 44 more
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[WARNING ] Detected JSESSIONID with invalid length; expected length of 23, found 24, setting: 31560cb7c4a6930415a4bb22 to null.
[AUDIT   ] CWPKI0817A: The default SSL configuration expects a <keyStore> element with an id value of defaultKeyStore, and a password. The defaultKeyStore <keyStore> element is missing, or the password is not specified. If SSL is not required, this message can be ignored. If SSL is required, either define the missing element: <keyStore id="defaultKeyStore" password="yourpassword" />, or change the default SSL configuration to use an existing keystore. See the following example: <ssl id="defaultSSLConfig" keyStoreRef="newKeyStore" />.
```

**Comment by denis-anisimov** [comment]
This is basically the same issue as https://github.com/vaadin/flow/issues/3920.

But here I don't even see how the production mode is enabled at all.....

- The server doesn't look started with the production mode property.
- Also I don't see the details of how the WAR has been built. Has it been built using maven plugin so that it's ready for the production?

**Comment by marcushellberg** [comment]
A customer reported the following error on Liberty 17 with the JEE7 feature. 

```
Exception = java.lang.NullPointerException
Source = com.ibm.ws.webcontainer.osgi.DynamicVirtualHost
probeid = startWebApp
Stack Dump = java.lang.NullPointerException
at com.vaadin.flow.server.startup.RouteRegistry.setErrorNavigationTargets(RouteRegistry.java:229)
at com.vaadin.flow.server.startup.ErrorNavigationTargetInitializer.onStartup(ErrorNavigationTargetInitializer.java:50)
at com.ibm.ws.webcontainer.webapp.WebApp.initializeServletContainerInitializers(WebApp.java:2487)
at com.ibm.ws.webcontainer.webapp.WebApp.initialize(WebApp.java:994)
at com.ibm.ws.webcontainer.webapp.WebApp.initialize(WebApp.java:6566)
at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost.startWebApp(DynamicVirtualHost.java:467)
at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost.startWebApplication(DynamicVirtualHost.java:462)
at com.ibm.ws.webcontainer.osgi.WebContainer.startWebApplication(WebContainer.java:1132)
at com.ibm.ws.webcontainer.osgi.WebContainer.access$000(WebContainer.java:108)
at com.ibm.ws.webcontainer.osgi.WebContainer$3.run(WebContainer.java:944)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at java.lang.Thread.run(Thread.java:748)
```

**Comment by Artur-** [comment]
@marcushellberg extracted the new NPE issue to #4511 as this issue is about production mode problems

**Comment by Artur-** [comment]
Can't reproduce this issue in Liberty 18 using Beverage Buddy with the fix for #4511 

Created war using
```
mvn clean install -PproductionMode
```

Deployed and checked that Chrome loads `frontend-es6/vaadin-flow-bundle-18ac7f3a43b5418.cache.html` instead of individual files
